pkg/workflow_execution.go

```diff
@@ -339,6 +341,31 @@ func (c *Client) CreateWorkflowExecution(namespace string, workflow *WorkflowExe
 		return nil, util.NewUserError(codes.NotFound, "Error with getting workflow template.")
 	}
 
+	if runtimeVars != nil && workflowTemplate.ArgoWorkflowTemplate != nil {
```

Comment:
I don't think this section should be here. It's unique to Workspaces only. Other calls to `CreateWorkflowExecution` may never have `"stateful-set-resource"` or `VirtualService`. Can we inject these into the template before it's passed in here?

Reply:
@rushtehrani I agree, we need to separate the concerns here.

The code above gets the workflow template and that also gets the k8s argo template. At the moment, we can't modify it beforehand because it gets it in this function.

We could change the function so that we pass in the workflow template with the argo contents pre-loaded.
That way it does less work and is more configurable. 

What do you think?

Reply:
@Vafilor parameters are showing and configuration reload is working. Is this the only change that remains?

Reply:
@rushtehrani Yes, I should have it ready by tomorrow morning.